packages/legend-query/src/components/QueryBuilderFunctionsExplorerPanel.tsx

```diff
@@ -0,0 +1,590 @@
+            </div>
+          </div>
+        </div>
+      }
+    >
+      <span>{children}</span>
+    </Tooltip>
+  );
+};
+
+const QueryBuilderFunctionDragLayer = observer(
+  (props: { queryBuilderState: QueryBuilderState }) => {
+    const { itemType, item, isDragging, currentPosition } = useDragLayer(
+      (monitor) => ({
+        itemType:
+          monitor.getItemType() as QUERY_BUILDER_FUNCTIONS_EXPLORER_TREE_DND_TYPE,
+        item: monitor.getItem<QueryBuilderFunctionsExplorerDragSource | null>(),
+        isDragging: monitor.isDragging(),
+        initialOffset: monitor.getInitialSourceClientOffset(),
+        currentPosition: monitor.getClientOffset(),
+      }),
+    );
+    if (
+      !isDragging ||
+      !item ||
+      !Object.values(QUERY_BUILDER_FUNCTIONS_EXPLORER_TREE_DND_TYPE).includes(
+        itemType,
+      ) ||
+      item.node.dndType ===
+        QUERY_BUILDER_FUNCTIONS_EXPLORER_TREE_DND_TYPE.PACKAGE
+    ) {
+      return null;
+    }
+    return (
+      <div className="query-builder__functions-explorer__tree__drag-preview-layer">
+        <div
+          className="query-builder__functions-explorer__tree__drag-preview"
+          style={
+            !currentPosition
+              ? { display: 'none' }
+              : {
+                  transform: `translate(${currentPosition.x + 20}px, ${
+                    currentPosition.y + 10
+                  }px)`,
+                }
+          }
+        >
+          {generateFunctionSignature(
+            item.node.packageableElement as ConcreteFunctionDefinition,
+            true,
+          )}
+        </div>
+      </div>
+    );
+  },
+);
+
+const QueryBuilderFunctionsExplorerListEntry = observer(
+  (props: {
+    queryBuilderState: QueryBuilderState;
+    element: ConcreteFunctionDefinition;
+    rootPackageName: ROOT_PACKAGE_NAME;
+  }) => {
+    const { queryBuilderState, element, rootPackageName } = props;
+    const node = generateFunctionsExplorerTreeNodeData(
+      queryBuilderState,
+      element,
+      rootPackageName,
+    );
+    const [, dragConnector, dragPreviewConnector] = useDrag(
+      () => ({
+        type: QUERY_BUILDER_FUNCTIONS_EXPLORER_TREE_DND_TYPE.FUNCTION,
+        item: { node: node },
+      }),
+      [node],
+    );
+    // hide default HTML5 preview image
+    useEffect(() => {
+      dragPreviewConnector(getEmptyImage(), { captureDraggingState: true });
+    }, [dragPreviewConnector]);
+
+    return (
+      <div
+        className="query-builder__functions-explorer__function"
+        ref={dragConnector}
+      >
+        <div className="query-builder__functions-explorer__function__content">
+          <div className="query-builder__functions-explorer__function__icon">
+            <div className="query-builder__functions-explorer__function-icon">
+              <PURE_FunctionIcon />
+            </div>
+          </div>
+          <div className="query-builder__functions__function__label">
```

Comment:
We need to do truncation here and potentially for the tree as well, when the text is too long, people can nolonger read it, we need to allow them to hover and read the full signature, so let's add `title={generateFunctionSignature(element, true)}` here, which means, we'd probably should create a constant to hold this so we don't need to call it twice.